aiida/calculations/arithmetic/add.py

```diff
@@ -0,0 +1,115 @@
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Implementation of CalcJobNode to add two numbers for testing and demonstration purposes."""
+from aiida import orm
+from aiida.common.datastructures import CalcInfo, CodeInfo
+from aiida.engine import CalcJob
+
+
+class ArithmeticAddCalculation(CalcJob):
+    """Implementation of CalcJob to add two numbers for testing and demonstration purposes."""
+
+    @classmethod
+    def define(cls, spec):
+        super().define(spec)
+        # yapf: disable
+        spec.inputs['metadata']['options']['parser_name'].default = 'arithmetic.add'
+        spec.inputs['metadata']['options']['input_filename'].default = 'aiida.in'
+        spec.inputs['metadata']['options']['output_filename'].default = 'aiida.out'
+        spec.inputs['metadata']['options']['resources'].default = {
+            'num_machines': 1, 'num_mpiprocs_per_machine': 1
+        }
+        spec.input('x', valid_type=(orm.Int, orm.Float), help='The left operand.')
+        spec.input('y', valid_type=(orm.Int, orm.Float), help='The right operand.')
+        spec.input('settings', required=False, valid_type=orm.Dict, help='Optional settings.')
+        spec.output('sum', valid_type=(orm.Int, orm.Float), help='The sum of the left and right operand.')
+        spec.exit_code(300, 'ERROR_NO_RETRIEVED_FOLDER',
+            message='The retrieved folder data node could not be accessed.')
+        spec.exit_code(310, 'ERROR_READING_OUTPUT_FILE',
+            message='The output file could not be read from the retrieved folder.')
+        spec.exit_code(320, 'ERROR_INVALID_OUTPUT',
+            message='The output file contains invalid output.')
+        spec.exit_code(410, 'ERROR_NEGATIVE_NUMBER',
+            message='The sum of the operands is a negative number. Only thrown if `settings.allow_negative = False`.')
+
+    def prepare_for_submission(self, folder):
+        """
+        This method is called prior to job submission with a set of calculation input nodes.
+        The inputs will be validated and sanitized, after which the necessary input files will
+        be written to disk in a temporary folder. A CalcInfo instance will be returned that contains
+        lists of files that need to be copied to the remote machine before job submission, as well
+        as file lists that are to be retrieved after job completion.
+
+        :param folder: an aiida.common.folders.Folder to temporarily write files on disk
+        :returns: CalcInfo instance
+        """
+        input_x = self.inputs.x
+        input_y = self.inputs.y
+        input_code = self.inputs.code
+
+        self.write_input_files(folder, input_x, input_y)
+
+        retrieve_list = self.get_retrieve_list()
+        local_copy_list = self.get_local_copy_list()
+        remote_copy_list = self.get_remote_copy_list()
```

Comment:
define these directly, i.e. without the methods and directly assign result to calcinfo, i.e. get rid of these lines and directly add to line 70 - 72